### ISSUE TITLE
R3BEventHeader registration disabled on purpose?

### ISSUE THREAD

**Issue by YanzhaoW** [R3BRootGroup/R3BRoot#1092]
**Describe the bug**

The `Register` virtual function of `R3BEventHeader` is overridden with empty function:

https://github.com/R3BRootGroup/R3BRoot/blob/56e3202b18a34097e0805236faba8759d650d62a/r3bbase/R3BEventHeader.h#L56-L58

This causes the `FairRunAna`, which calls the `register` function automatically in the `init` function, fail to register the event header to the `FairRootManager`. It also leads to the result `FairRootManager::GetObject` nullptr if the tasks try to obtain the event header with `GetObject` function.

I don't know whether this is done on purpose or it's just a bug? 

If it's a bug, I suggest to remove the line above.

**Comment by jose-luis-rs** [comment]
Yes, I guess that happens because the R3BEventHeader was not registered in the analysis macro:

// Create run  --------------------------------------------
auto *run = new FairRunAna();
// Set up R3BHeader  -----------------------------------
auto *EvntHeader = new R3BEventHeader();
EvntHeader->SetExpId(fExpId);
run->SetEventHeader(EvntHeader);

Do you know how to improve this part?

**Comment by YanzhaoW** [comment]
Hi,

I would say just remove this line of code. In this way, the `Register` will not be overridden and the original definition from the base class `FairEventHeader` will be used. 

Is there any particular reason `R3BEventHeader` **must not** be registered for FairRunAna?

**Comment by jose-luis-rs** [comment]
Hello

For simulations we don't need the R3BEventHeader

**Comment by YanzhaoW** [comment]
Hi,

But FairRunAna is not for simulations. Instead FairRunSim is for simulations.

**Comment by jose-luis-rs** [comment]
This depends on how one wants to use the code. FairRunSim is for simulations and to generate the simulated data, but here one can select different options:
1) Use FairRunSim to generate the simulated data for all levels: point, Cal, Hit, ...
2) Use FairRunSim to generate the simulated data up to the point level and then FairRunAna to move data from point to Cal, Hit and other analysis tasks.

**Comment by YanzhaoW** [comment]
Hi,

Yes, what you said is correct. But the point is that `FairRunAna` is for the data analysis, which converts one data to another. The data it converts could be from simulation, could also be from calibration and could be anything.

But I don't think it's correct to say because `FairRunAna` could be used for simulation and simulation doesn't need `FairEventHeader`, `FairRunAna` doesn't need the event header.

**Comment by jose-luis-rs** [comment]
I use FairRunAna for exp. data analysis but also for simulated data to test the reconstruction algorithms because this way usually reduces the computing time. 

"FairRunAna doesn't need the event header" -> Ok, but we need the event header to get the information about the triggers, tpats, timestamp, etc. So if the event header is not there we will miss this very important information for data analysis. 

If we use FairRunOnline for unpacking data up to the Cal level and then we want to produce the Hit level without running the code again over the unpacking (just to reduce the computing time), we must use FairRunAna to move the data and this needs to propagate the event header as well. 

In the case of simulation, the propagation of the event header (FairMCEventHeader) is not so important, but I don't know if this could be important in the future ...

**Comment by YanzhaoW** [comment]
Close this issue for now.
